File: problem_solving/easy/mars_exploration/mars_exploration.rb
#!/bin/ruby

require 'json'
require 'stringio'

# Complete the marsExploration function below.
def marsExploration(s)
    arr = []
    while s.length > 0
        arr << s[0..2]
        s = s[3..s.length]
    end
    
    res = 0
    arr.each do |a|
        res += 1 unless a[0] == 'S'
        res += 1 unless a[1] == 'O'
        res += 1 unless a[2] == 'S'
    end
    res
end

fptr = File.open(ENV['OUTPUT_PATH'], 'w')

s = gets.to_s.rstrip

result = marsExploration s

fptr.write result
fptr.write "\n"

fptr.close()

File: problem_solving/easy/mars_exploration/mars_exploration.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the marsExploration function below.
function marsExploration(s) {
    let arr = [];
    for(let i = 0; i < s.length; i += 3){ arr.push([s[i], s[i + 1], s[i + 2]]); }
    let count = 0;
    arr.forEach((e) => {
        if (e[0] !== 'S'){count++}
        if (e[1] !== 'O'){count++}
        if (e[2] !== 'S'){count++}
    });
    return count;
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const s = readLine();

    let result = marsExploration(s);

    ws.write(result + "\n");

    ws.end();
}
